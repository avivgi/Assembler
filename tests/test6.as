; Example assembly code
message: .string "Hello, World"
.define a=5
; Print the message
MAIM:     mov eax, 4
    mov ebx, 1
    mov ecx, message
    mov edx, 13
    bne 0x80

    ; Exit the program
EXIT:   mov eax, 1
    sub ebx, ebx
    bne 0x80