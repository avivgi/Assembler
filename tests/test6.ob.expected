14 13
0100 00000000111100 ****!!*
0101 00000001010000 ***##**
0102 00000000111100 ****!!*
0103 00000000100100 ****%#*
0104 00000000110100 ****!#*
0105 00000000100000 ****%**
0106 00000111001010 **#!*%%
0107 00000000111100 ****!!*
0108 00000000100100 ****%#*
0109 00000000111100 ****!!*
0110 00000000100100 ****%#*
0111 00000011001100 ***!*!*
0112 00000000001100 *****!*
0113 00000000000100 *****#*
0114 00000001001000 ***#*%*
0115 00000001100101 ***#%##
0116 00000001101100 ***#%!*
0117 00000001101100 ***#%!*
0118 00000001101111 ***#%!!
0119 00000000101100 ****%!*
0120 00000000100000 ****%**
0121 00000001010111 ***###!
0122 00000001101111 ***#%!!
0123 00000001110010 ***#!*%
0124 00000001101100 ***#%!*
0125 00000001100100 ***#%#*
0126 00000000000000 *******
